### PR TITLE
DevEx - transition some workflows to GitHub actions

### PR DESCRIPTION
**Pull request by codyseibert**
Refactoring all pre-deploy checks from circle into github actions.  This refactor should reduce our ci/cd pipeline costs (github actions states they are free for public repo) and also help simplify our process as we hone in on a single service for builds.

requires a docker-to-ecr run on the court environments

### CODE REVIEW

**Review by rgoshen — APPROVED** [review]

**Review by TomElliottFlexion — APPROVED** [review]

**Review by rachelschneiderman — COMMENTED** [review]
This is awesome! I noticed it's configured right now to run on `test` and `dependency-updates` branches, should `prod` and `staging` at least be added for now?

**Comment by codyseibert** [comment]
planning to add the circle ci branch tests back until we transition everything into actions

File: .circleci/config.yml [hunk 1]
```diff
@@ -61,378 +61,9 @@ commands:
             - node_modules
 
 jobs:
-  build-client-integration:
-    docker:
-      - image: $AWS_ACCOUNT_ID.dkr.ecr.us-east-1.amazonaws.com/ef-cms-us-east-1:2.12.0
-        aws_auth:
-          aws_access_key_id: $AWS_ACCESS_KEY_ID
```
Comment (mmarcotte) [inline .circleci/config.yml]: While I love the cleanup, I think we will need these tests for the merged code on the `test` branch/environment. Can you reimplement those for that branch/env?

File: Dockerfile [hunk 2]
```diff
@@ -24,6 +24,7 @@ RUN apt-get install -y -t \
   graphicsmagick=1.4+really1.3.36+hg16481-2 \
   ghostscript=9.53.3~dfsg-7+deb11u2 \
   chromium \
+  sudo \
```
Comment (mmarcotte) [inline Dockerfile:29]: does this necessitate building a new docker tag and pushing it to ecr?

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**